scripts/ci/benchmarks-report.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+# Takes raw reports from
+# "cargo bench --bench benches -- --noplot --save-baseline master" output as 1st argument
+# "cargo bench --bench benches -- --noplot --baseline master" output as 2nd argument
+# Parses them to json and posts formatted results to PR on a GitHub as a comment
+set -eu
+set -o pipefail
+
+PR_COMMENTS_URL="https://api.github.com/repos/paritytech/wasmi/issues/${CI_COMMIT_BRANCH}/comments"
+
+# master report to json
+echo "PARSING MASTER REPORT"
+sed -e 's/^Found.*//g' \
+    -e 's/^\s\+[[:digit:]].*$//g' \
+    -e 's/\//_/g' \
+    -e 's/^[a-z0-9_]\+/"&": {/g' \
+    -e 's/time:\s\+\[.\{10\}/"time": "/g' \
+    -e 's/.\{10\}\]/"},/g' \
+    -e '1s/^/{\n/g' \
+    -e '/^$/d' \
+    -e 's/  */ /g' \
+    -e 's/^ *\(.*\) *$/\1/' $1 \
+    | sed -z 's/.$//' \
+    | sed -e '$s/.$/}/g' \
+    | tee target/criterion/output_master.json
```

Comment:
As a follow up to this PR we should try to make this part more readable.
For example criterion supports JSON output as well: https://bheisler.github.io/criterion.rs/book/cargo_criterion/external_tools.html
And with tools such as `jq` we could significantly make extracting information more readable.

I will merge this PR and propose a follow-up PR for you. :) Is that okay with you?

Reply:
Agree. My homegrown parsing "engine" looks ugly. And definitely can be improved.
But json export tool you mention is a part of `cargo-criterion` which is under development. By the way, have you tried it in action?
Reports from `cargo bench` also produces jsons with raw data in `target/criterion`. I've tried to use them as well, but found that even it is possible get execution times and other metrics. Only thing stopped me from using it is that I forced to write my own statistics analysis tool to interpret them :)
So parsing raw command line output to json appeared easier solution.

Reply:
Upd: tried to install and use `cargo-criterion` and first attempt was not successful:
```
Benchmarking compile_and_validate/v1thread 'main' panicked at 'could not open benchmark file benches/wasm/wasm_kernel/target/wasm32-unknown-unknown/release/wasm_kernel.wasm: No such file or directory (os er',  wasmi_v1/benches/bench/mod.rs : 16 : 33 
 note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Benchmarking compile_and_validate/v1: Warming up for 1.0000 sthread 'main' panicked at 'Unexpected message FinishedBenchmarkGroup { group: "compile_and_validate/v1" }', /home/sergej/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-criterion-1.1.0/src/bench_target.rs:306:26
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Io(Error { kind: UnexpectedEof, message: "failed to fill whole buffer" })', /home/sergej/.cargo/registry/src/github.com-1ecc6299db9ec823/criterion-0.4.0/src/benchmark_group.rs:380:18
stack backtrace:
```

Double checked `benches/wasm/wasm_kernel/target/wasm32-unknown-unknown/release/wasm_kernel.wasm` is in place. What could be the reason for failure.

Reply:
The problem with raw command line output compared to the JSON output (or genreally machine readable output) is that the former is subject to change whereas the latter is kinda guaranteed to be stable to not break dependents. Therefore as a long term non-brittle solution we really really want to read output that is intended to be machine readable.

Reply:
> Upd: tried to install and use `cargo-criterion` and first attempt was not successful:
> 
> ```
> Benchmarking compile_and_validate/v1thread 'main' panicked at 'could not open benchmark file benches/wasm/wasm_kernel/target/wasm32-unknown-unknown/release/wasm_kernel.wasm: No such file or directory (os er',  wasmi_v1/benches/bench/mod.rs : 16 : 33 
>  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> Benchmarking compile_and_validate/v1: Warming up for 1.0000 sthread 'main' panicked at 'Unexpected message FinishedBenchmarkGroup { group: "compile_and_validate/v1" }', /home/sergej/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-criterion-1.1.0/src/bench_target.rs:306:26
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Io(Error { kind: UnexpectedEof, message: "failed to fill whole buffer" })', /home/sergej/.cargo/registry/src/github.com-1ecc6299db9ec823/criterion-0.4.0/src/benchmark_group.rs:380:18
> stack backtrace:
> ```
> 
> Double checked `benches/wasm/wasm_kernel/target/wasm32-unknown-unknown/release/wasm_kernel.wasm` is in place. What could be the reason for failure.

You are probably doing this in the root directory instead of the `wasmi_v1` subdirectory.

Reply:
You are right. Now it works, at first glance json's produced by `cargo-criterion` are almost what is needed. Of course some processing also is needed to produce readable output, like time conversion from ns into more convenient units. But at least we'll have an out-of-the-box machine readable data